docker/hooks/build

```diff
@@ -0,0 +1,35 @@
+#!/bin/bash
+
+# navigate from Dockerfile folder to project root
+cd ..
+
+### upgrade to Docker 18 ###
+docker --version
+apt-get update && apt-get dist-upgrade -y && apt-get install jq -y
+docker --version
+mkdir ~/.docker
+AUTHTOKEN=$(cat ~/.dockercfg | jq '.["https://index.docker.io/v1/"] .auth')
+echo "{\"auths\":{\"https://index.docker.io/v1/\":{\"auth\": $AUTHTOKEN}},\"experimental\": \"enabled\"}" > ~/.docker/config.json
+### -------------------- ###
+
+MULTIARCH_DOCKERFILE_PATH="${DOCKERFILE_PATH/Dockerfile/multiarch.Dockerfile}"
+
+# register qemu modules
+docker run --rm --privileged multiarch/qemu-user-static:register
+
+# remove MULTIARCH line and build default image
+sed "/__MULTIARCH_/d" "$MULTIARCH_DOCKERFILE_PATH" > "$DOCKERFILE_PATH"
+docker build --build-arg BUILD_GOOS="linux" --build-arg BUILD_GOARCH="amd64" --build-arg SERVE_U="amd64" --build-arg SERVE_R="alpine" --build-arg QEMUARCH="amd64" -f "$DOCKERFILE_PATH" -t "${IMAGE_NAME/:latest/:amd64-latest}" .
+
+# enable MULTIARCH line and build multiarch images
+sed "s/__MULTIARCH_//g" "$MULTIARCH_DOCKERFILE_PATH" > "$DOCKERFILE_PATH"
+curl -L "https://github.com/multiarch/qemu-user-static/releases/download/v3.1.0-2/x86_64_qemu-i386-static.tar.gz" | tar xz
+docker build --build-arg BUILD_GOOS="linux" --build-arg BUILD_GOARCH="386" --build-arg SERVE_U="i386" --build-arg SERVE_R="alpine" --build-arg QEMUARCH="i386" -f "$DOCKERFILE_PATH" -t "${IMAGE_NAME/:latest/:i386-latest}" .
+curl -L "https://github.com/multiarch/qemu-user-static/releases/download/v3.1.0-2/x86_64_qemu-arm-static.tar.gz" | tar xz
```

Comment:
For some sort of maintainability, I'd really like to see this as a loop over archs with the eqmu version broken out into a variable. As is this is a wall of magic, which we will need to get into to change stuff in the future...

Reply:
Yes, I thought about this too, but I weren't able to come up with a nice way to store the build argument combination for easy looping as I'm not incredibly experienced with bash. Could you do implement this?

Reply:
I admit I'm a bit damaged and treating bash as a real language only goes so far, but something like

```
archs=( 386:i386:i386 arm:arm32v6:arm ... )
for arch in "${archs[@]}" ; done
  parts=( ${arch//:/ } )
  goarch=${parts[0]}
  serve_u=${parts[1]}
  qemuarch=${parts[2]}
  docker build ...
done
```